pl_bolts/datasets/base_dataset.py

```diff
@@ -58,3 +60,31 @@ def _download_from_url(self, base_url: str, data_folder: str, file_name: str):
             urllib.request.urlretrieve(url, fpath)
         except HTTPError as err:
             raise RuntimeError(f"Failed download from {url}") from err
+
+
+@dataclass
+class DataModel:
+    """Data model dataclass.
+
+    Ties together data and callable transforms.
+
+    Attributes:
+        data: Sequence of indexables.
+        transform: Callable to transform data.
+    """
+
+    data: ARRAYS
+    transform: Optional[Callable] = None
```

Comment:
The `transform` can be better typed:

```suggestion
    transform: Optional[Callable[[ARRAYS], ARRAYS]] = None
```